### DISCUSSION TITLE
Allow changing the current device from the left-click pop-up

### DISCUSSION THREAD

**Discussion by vertigo220** [File-New-Project/EarTrumpet#709] [Ideas]
This would make it more consistent with the standard Win10 volume control.

**Comment by riverar** [comment]
We support this via right-click and don't have any plans to change the current flyout UI. I'll keep this open though as there may be changes in the next version of Windows that force us to revisit this UI.

**Comment by vertigo220** [comment]
I realize that, I just think it would be better to allow it in the left-click pop-up, since a) that's consistent with the Windows volume control, and b) it allows everything to be done in one interface, instead of having to open one, then click somewhere to close it, click the arrow to expand notifications (I keep volume control hidden, as I don't want it using up taskbar space), then clicking again on the icon to open the other menu. All it would take is a button next to each audio device to set that device as the default.

**Comment by riverar** [comment]
I hear ya.

But consider it takes 3 clicks to change the default device with EarTrumpet and 5 with the normal flyout. Your scenario seems easier and faster to execute in EarTrumpet. Am I missing something?

Normal
* Left-click overflow chevron
* Left-click icon
* Left-click device selection area
* Left-click new audio device
* Left-click the desktop to dismiss the flyout

EarTrumpet
* Left-click overflow chevron
* Right-click EarTrumpet icon
* Left-click new audio device

**Comment by vertigo220** [comment]
Two things to keep in mind, though. First, I'm not saying to _replace_ the right-click method of switching, just allow switching from the left-click menu as well. Because even though the right-click method is more efficient, it's less efficient if you _have_ to do it that way when you're already in the left-click flyout, whether due to accidentally left-clicking due to being used to Windows' default behavior, or because you not only want to change the default device, but other settings as well.

Second, it would only require 4 clicks: left-click chevron, left-click Trumpet, left-click button next to audio device you want as default, and left-click elsewhere to close (which you'd likely be doing anyway). So still more efficient than Windows, and only slightly less efficient than the current right-click method. Granted, if I remember to use right-click, that would be my preferred method if that's all I wanted to do, because it is _much_ easier (not just the one less click, but a small pop-up with a few entries to quickly find what you want vs the massive one with all the apps), but I see no reason not to _also_ have the ability in the left-click menu, and at least two reasons to have it.

**Comment by GoldenTao** [comment]
Thanks for the feedback, how would you see this working in EarTrumpet? Our expand to multi-device UI currently replaces where the original Windows UI has its device changer dropdown (don't get me started on the UX nightmare that is the Windows Volume flyout.)  

It's not something we're looking to change as @riverar said and we would not be able to replicate the same UI as the original Windows dialog with how our flyout works but if you have an interesting idea as to how this might be implemented we can have this thread handy in case something changes.

**Comment by vertigo220** [comment]
I'm not suggesting replicating it. How I envision it is when you left-click the icon to open the flyout, then click the audio device to expand it to show all devices. Then the devices would have an icon next to them, and you would click the icon to make that device the current/default one, which should also change the icon, so maybe it would just be a speaker playing sound, similar, though not too similar, to the toggle mute icon, which would change from greyed-out to blue or something. Or you could double-click the device name, which would change its color to indicate that it's set as default (or the mute icon right below it could change), or right-click and set as default (though that's the least ideal option). The problem with double-clicking is that wouldn't work on the top device, since clicking that one collapses the flyout. Another option would be to right-click the mute button, then either show a context-menu to set the device as default or just have it happen when right-clicking that, or when right-clicking the device name. Lots of possibilities, it's just a matter of deciding which one is the best combination of working well, being intuitive, and being easy, i.e. less clicks. As with anything, probably best to gather feedback from users on what they feel would be best. Another advantage to this is that by adding the ability to change the default device in the flyout, that would necessitate having it actually show which one is currently set as default, something it currently doesn't do. For example, if changing the default changes the device name or its mute icon to a different color or even just changes that icon to be underlined or encircled or something, then that would also allow the user to see which device is the current default while in the flyout, something that, again, currently can't be done.
